extensions/agroal/runtime/src/main/java/io/quarkus/agroal/metrics/AgroalCounter.java

```diff
@@ -0,0 +1,81 @@
+package io.quarkus.agroal.metrics;
+
+import javax.enterprise.inject.spi.CDI;
+
+import org.eclipse.microprofile.metrics.Counter;
+
+import io.agroal.api.AgroalDataSource;
+import io.agroal.api.AgroalDataSourceMetrics;
+
+public class AgroalCounter implements Counter {
+
+    private String dataSourceName;
+    private AgroalDataSource dataSource;
+    private String metric;
+
+    public AgroalCounter() {
+
+    }
+
+    /**
+     * @param dataSourceName Which datasource should be queried for metric
+     * @param metricName Name of the method from DataSource.getMetrics() that should be called to retrieve the particular value.
+     *        This has nothing to do with the metric name from MP Metrics point of view!
+     */
+    public AgroalCounter(String dataSourceName, String metricName) {
+        this.dataSourceName = dataSourceName;
+        this.metric = metricName;
+    }
+
+    public String getDataSourceName() {
+        return dataSourceName;
+    }
+
+    public void setDataSourceName(String dataSourceName) {
+        this.dataSourceName = dataSourceName;
+    }
+
+    private AgroalDataSource getDataSource() {
+        if (dataSource == null) {
+            if (dataSourceName == null) {
+                dataSource = CDI.current().select(AgroalDataSource.class).get();
+            } else {
+                dataSource = CDI.current().select(AgroalDataSource.class, new DataSourceLiteral(dataSourceName)).get();
```

Comment:
Inside extensions will usually direclty use `Arc` instead of `CDI`.
Not sure it have a performance impact.

Reply:
Ok, changed that to Arc